app/javascript/app/document-capture/components/form-steps.jsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+function FormSteps({ steps, onComplete }) {
+  const [values, setValues] = useState({});
+  const [stepIndex, setStepIndex] = useState(0);
+
+  function setStepValue(name, nextStepValue) {
+    setValues({ ...values, [name]: nextStepValue });
+  }
+
+  /**
+   * Increments state to the next step, or calls onComplete callback if the
+   * current step is the last step.
+   */
+  function toNextStep() {
+    const nextStepIndex = stepIndex + 1;
+    const isComplete = nextStepIndex === steps.length;
+    if (isComplete) {
+      // Clear step parameter from URL.
+      window.history.pushState(null, null, window.location.pathname);
+
+      onComplete(values);
+    } else {
+      const { name: nextStepName } = steps[nextStepIndex];
+
+      // Push the next step to history, both to update the URL, and to allow the
+      // user to return to an earlier step (see `popstate` sync behavior).
+      window.history.pushState(
+        { stepIndex: nextStepIndex },
```

Comment:
If every step has a name (and we use the name in the URL), do we need the integer index of the step still? Could we just keep using the step names and do a `.find((step) => step.name == name)` kind of thing? Do we expect to have multiple steps with the same name?

Reply:
> If every step has a name (and we use the name in the URL), do we need the integer index of the step still?

Not necessarily, no. There's an `O(n)` (search) vs. `O(1)` (access) performance difference, though in practice this would most certainly prove to be negligible here.

It could have some benefit to track only the name, since we could consolidate the source of truth to the value derived from the URL. Having a single value to track would also make it more generic and available for reuse, which relates to a point I raised in an [earlier comment](https://github.com/18F/identity-idp/pull/3952#discussion_r457608657):

>In the usage here, it would likely require some mapping back and forth between the two values "step" and "name", since useHistoryParam would ideally only need to support one value (the query parameter key).

Reply:
Yeah my thinking was similar, there are likely going to be a small enough list of steps that the search access is not a huge deal -- I do think it might be worth it to simplify for now?